src/blitter/common.hpp

```diff
@@ -192,4 +192,24 @@ void Blitter::DrawLineGeneric(int x1, int y1, int x2, int y2, int screen_width,
 	}
 }
 
+template <typename T>
+/* static */ void Blitter::MovePixels(const T *src, T *dst, size_t width, size_t height, size_t pitch)
```

Comment:
All the callers have this `if` - `else` branch depending on scrolling up/down.
Can these now be merged?

Reply:
The `if`-branch iterates the rows in the opposite order, so it wouldn't work out of the box. Though, if I make `pitch` a `ssize_t` it ought to work.